tools/src/icon4py/tools/py2fgen/wrappers/common.py

```diff
@@ -165,14 +165,14 @@ def construct_icon_grid(
     log.debug("Offsetting Fortran connectivitity arrays by 1")
 
     xp = data_alloc.import_array_ns(backend)
-
-    cells_start_index = adjust_fortran_indices(cell_starts)
-    vertex_start_index = adjust_fortran_indices(vertex_starts)
-    edge_start_index = adjust_fortran_indices(edge_starts)
-
-    cells_end_index = cell_ends
-    vertex_end_index = vertex_ends
-    edge_end_index = edge_ends
+    # TODO (halungge): icon has 0 values in these arrays in some places possibly where they don't use them in case
```

Comment:
See my comment above in `serialbox.py`. Whenever the values are `> 0` the denote real indices, but there are 0 values which probably denote "invalid". Generally adjusting them to 0 based Python indices might be a source of errors.